tests/test_objective_funs.py

```diff
@@ -485,25 +493,284 @@ def test_boundary_error_nestor(self):
+        # part of the equilibrium (Rb_lmn) since field is axisymmetric.
+        field = axisym_field.copy()
+        eq = nonaxisym_eq.copy()
+        field_grid, eval_grid, source_grid = get_grids(eq, field)
+
+        # fix modes where Rb_lmn = 0 so optimizer only pays attention to
+        # nonaxisymmetric modes
+        zero_indices = np.where(eq.Rb_lmn == 0)
+        fixed_R_modes = eq.surface.R_basis.modes[zero_indices]
+
+        zero_indices = np.where(eq.Zb_lmn == 0)
+        fixed_Z_modes = eq.surface.Z_basis.modes[zero_indices]
+
+        constraints = (
+            ForceBalance(eq),
+            FixPressure(eq),
+            FixCurrent(eq),
+            FixPsi(eq),
+            FixBoundaryR(eq, modes=fixed_R_modes),
+            FixBoundaryZ(eq, modes=fixed_Z_modes),
+        )
+        quadflux_obj = QuadraticFlux(
+            ext_field=field,
+            eq=eq,
+            src_grid=source_grid,
+            eval_grid=eval_grid,
+            field_grid=field_grid,
+            field_fixed=True,
+        )
+        objective = ObjectiveFunction(quadflux_obj)
+
+        things_with_field_fixed, __ = nonlinear_optimizer.optimize(
+            eq,
+            objective=objective,
+            constraints=constraints,
+            gtol=0,
+            ftol=0,
+            copy=True,
+        )
+
+        new_Rb_lmn = things_with_field_fixed[0].Rb_lmn[
+            np.where(eq.surface.R_basis.modes[:, 2] != 0)
+        ]
+        new_Zb_lmn = things_with_field_fixed[0].Rb_lmn[
+            np.where(eq.surface.R_basis.modes[:, 2] != 0)
+        ]
+
+        np.testing.assert_allclose(new_Rb_lmn, 0, atol=1e-10)
+        np.testing.assert_allclose(new_Zb_lmn, 0, atol=1e-10)
+
+        # test with eq_fixed = field_fixed = False: nonaxisymmetric field and
+        # nonaxisymmetric equilibrium
+        eq = Equilibrium(M=2, L=2, N=4)
+        field = make_nonaxisym_field(eq)
+        field_grid, eval_grid, source_grid = get_grids(eq, field)
+        surface_grid = LinearGrid(
+            rho=np.array([1.0]),
+            M=20,
+            N=20,
+            NFP=int(eq.NFP),
+            sym=False,
+        )
+
+        quadflux_obj = QuadraticFlux(
+            ext_field=field,
+            eq=eq,
+            src_grid=source_grid,
+            eval_grid=eval_grid,
+            field_grid=field_grid,
+        )
+        plasma_vessel_obj = PlasmaVesselDistance(
+            surface=field,
+            eq=eq,
+            plasma_grid=source_grid,
+            surface_grid=surface_grid,
+            bounds=(0.2, 2),
+        )
+        mean_curvature_obj = MeanCurvature(eq, bounds=(-3, 3), grid=surface_grid)
+        objective = ObjectiveFunction(
+            (quadflux_obj, plasma_vessel_obj, mean_curvature_obj)
+        )
+
+        constraints = (
+            ForceBalance(eq),
+            FixPressure(eq),
+            FixCurrent(eq),
+            FixPsi(eq),
+            FixParameter(field, ["I", "G", "R_lmn", "Z_lmn"]),
```

Comment:
This constraint shouldn't be necessary for this test, since we don't want to force the field to be axisymmetric. Maybe fixing the surface is helpful, but at least remove the constraints on `"I"` and `"G"`

Reply:
if we remove those constraints, then the minimum Bnormal will be just making `I` and `G` and `Phi_mn` equal to zero, which is not what we are trying to test here